src/main/java/org/bobocode/CustomSession.java

```diff
@@ -14,7 +13,7 @@ public class CustomSession {
     private final Map<String, Session> sessionStore = new ConcurrentHashMap<>();
 
     public Session createOrGetSession(HttpServletRequest request, HttpServletResponse response) {
-        if (Objects.requireNonNull(getSessionId(request)).isEmpty()) {
+        if (getSessionId(request) == null) {
             var session = createCustomSession(request);
             response.addCookie(new Cookie(Session.SESSION_ID, session.getSessionId()));
```

Comment:
what if we  merge `getSessionId` method and `createCustomSession` method and line 18 which all work with cookie, and if it absents at cookies we call create a session and set it to the cookie without a name (default null value), and after that, we call `getCustomSession` which already implement logic with put name if it changed, so our null will override with parameter from request. Also check possible null pointer on line 31 same as you tried do in `createCustomSession`